### ISSUE TITLE
For statblocks with minor problems, handle exceptions where possible and do partial imports in the case that exceptions cannot be handled.

### ISSUE THREAD

**Issue by adventuremagic123** [Lavaeolous/PF1-StatBlock-Converter-Module#295]
**Summary:**  

Exceptions in imperfect statblocks should be handled wherever possible, because due to the frequency this is a problem, people may otherwise give up on using this module.  Worst case should be to simply not import the part of the statblock that cannot be parsed -- but still import the NPC or PC with warning messages.

**Details:**

Keep in mind that most of the audience for this tool (module) does not have a programming background -- and (separate issue) does not know what a "standard" statblock should look like.

So, for example, in #285 (Can't import Bestiary - Rakshasa ...) the comment was the following for closing this issue:

**_The HeroLab one works, the others are missing the Perform Type, e.g. "Perform +16" needs to be "Perform (Sing) +16"._**

There's some problems with this:

0. The Bestiary was used to help test this module.  The reality is that GMs still need to import NPCs for the Adventure Path they are using in their campaign -- even though the Compendium has almost everything from all the Bestiaries.  It's also not desirable to try to use the Compendium while running a game because it's just too slow.

1. That statblock was chosen as a regression test -- most of the Bestiary was not retested.  This means many NPCs might not now import because they wrote the Bestiary this way.

2. Most GMs probably won't care if "Perform" imports at all -- but would very much like to have the NPC import because they need it for their game.  The alternative of manually inputting the information by manually creating the NPC is very not desirable.

One choice for this case would be to do some exception handling and import this as "Perform (Unknown)" or just don't import it but warn the GM.

3. If this module breaks on imperfect statblocks often and doesn't create an NPC at all -- eventually people will give up on it.

**Comment by Lavaeolous** [comment]
So, for the example #285 given:
After closing that one, i created #290, as that is the underlying reason for #285 to fail. When #290 is implemented, the issues you noted should resolve itself, as all statblocks with undefined subskills for craft/perform/etc. will parse.

For the overlaying issue of usable feedback before starting a conversion thats doomed to fail, see issue #4 .

Work on that has started, see an early iteration here: https://imgur.com/VaHefhd

**Comment by adventuremagic123** [comment]
Ok.  I see.  Sounds good.  It goes without saying that no one will ever fix these statblocks.  Oops!  I said it.  :)

**Comment by Lavaeolous** [comment]
v2.0.0 is up and includes the preview. Next are more meaningful error messages.
